### PR TITLE
[Subtitles] Bump libass to 0.15.1

### PR DESCRIPTION
**Pull request by enen92**
## Description
This PR updates the libass version to 0.15.1 which was released yesterday as a bugfix release. Full changelog is available here: https://github.com/libass/libass/releases/tag/0.15.1. It fixes extraction of bundled fonts in ASS files for our use of the libass API: https://github.com/libass/libass/issues/266
Since 0.15.0, harfbuzz is a required dependency so it is included as well.

@lrusak can you upload the required sources to the mirrors? Also I am not sure what needs to be done for the linux specific guides because of harfbuzz

@Paxxi if not asking for too much can you adjust the windows version (https://github.com/xbmc/xbmc/pull/19561) to use 0.15.1 instead of 0.15.0 and upload new packages?

## Motivation and context
Keep up with current versions. Fix bundled ASS fonts.

## How has this been tested?
Tested with my ASS samples in OSX. Additionally tested the sample provided here: https://code.videolan.org/videolan/vlc-winrt/-/issues/265

## What is the effect on users?
Better support for ASS files. Better shaper for complex fonts.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: tools/depends/target/harfbuzz/Makefile [hunk 1]
```diff
@@ -0,0 +1,61 @@
+include ../../Makefile.include
+DEPS= ../../Makefile.include Makefile
+
+# lib name, version
+LIBNAME=libharfbuzz
+VERSION=2.8.0
+BASE_URL = https://github.com/harfbuzz/harfbuzz/archive/refs/tags
```
Comment (lrusak) [inline tools/depends/target/harfbuzz/Makefile:7]: todo: remove when uploaded to mirrors

File: tools/depends/target/libass/Makefile [hunk 2]
```diff
@@ -3,9 +3,11 @@ DEPS= ../../Makefile.include Makefile
 
 # lib name, version
 LIBNAME=libass
-VERSION=0.14.0
-SOURCE=$(LIBNAME)-$(VERSION)
-ARCHIVE=$(SOURCE).tar.gz
+VERSION=0.15.1
+#SOURCE=$(LIBNAME)-$(VERSION)
+#ARCHIVE=$(SOURCE).tar.gz
+BASE_URL=https://github.com/libass/libass/releases/download/$(VERSION)/
```
Comment (lrusak) [inline tools/depends/target/libass/Makefile:9]: todo: remove when uploaded to mirrors

File: tools/depends/target/harfbuzz/Makefile (hunk 1 above)
Comment (lrusak) [reply tools/depends/target/harfbuzz/Makefile:7]: uploaded

File: tools/depends/target/libass/Makefile (hunk 2 above)
Comment (lrusak) [reply tools/depends/target/libass/Makefile:9]: uploaded

**Comment by enen92** [comment]
Jenkins build this please

**Comment by Paxxi** [comment]
I got a surprise task at work so next two weeks will be bad but I'll try and look at it when possible.

Feel free to merge this without the windows 1.5.1 bump for now.

**Comment by enen92** [comment]
No problem paxxi, it's not like v20 will be out in 2 weeks :)

**Comment by lrusak** [comment]
@enen92 this PR should be fine as is, the windows libs can be added in a future PR

**Comment by enen92** [comment]
Sure. Either way Jenkins seems to build draft prs as well :)

**Review by lrusak — APPROVED** [review]
LGTM

**Pull request merged**